lib/serverengine/supervisor.rb

```diff
@@ -44,6 +44,14 @@ def initialize(server_module, worker_module, load_config_proc={}, &block)
       @enable_detach = !!@config[:enable_detach]
       @exit_on_detach = !!@config[:exit_on_detach]
       @disable_reload = !!@config[:disable_reload]
+
+      @command_pipe = @config.fetch(:command_pipe, nil)
```

Comment:
Ah, I did not make the mistake :p
The purpose of `@command_pipe` (and the `:command_pipe` option) is described as above.